Update format function to mention list, map, and null types (#31351)
hashicorp/terraform: website/docs/language/functions/format.mdx

format(spec, values...)
```

> In addition to `string`, `bool`, and `number`, you can use `format` to convert items of `null`, `list`, and `map` types into a formatted string.

Author: ```suggestion
You can use `format` to convert all [Terraform language types](/language/expressions/types#types) into a formatted string, including items of `null`, `list`, and `map` types.
```

> 

Reviewer: It feels a little "teasing" to say that it's possible to do it without saying anything about _how_ to do it. :thinking: 

For other functions I think we've typically put this sort of content under the "Examples" heading so that we can both show that something is possible and show an example of achieving it at the same time. For this case I think what this sentence is talking about is that the `%v` and `%#v` "verbs" will both accept a value of any type, always choosing _some_ reasonable way to show that value and never producing an "unsupported type" error, whereas most of the other "verbs" have some restrictions on what values they can show such as `%d` producing an error if the corresponding argument cannot convert to a number.

Author: This is fair - I think I was thinking that we could do something similar to what we did on the custom conditions page recently, where we just call out that this is possible. https://www.terraform.io/language/expressions/custom-conditions#error-messages

I think what I may do here is update the descriptions for those verbs themselves to show that they accept values of any type, will convert that value into a formatted string, and will not error.  Do you think it's appropriate to also include an example of using them to do that, or do we think that's too much here? I want to walk a line between giving all the information but also not over explaining to the folks who are using this (and annoying them).
